langchain/src/vectorstores/elasticsearch.ts

```diff
@@ -0,0 +1,172 @@
+import * as uuid from "uuid";
+import {
+  BulkOperationContainer,
+  BulkRequest,
+} from "@elastic/elasticsearch/lib/api/types.js";
+import { Client, ClientOptions } from "@elastic/elasticsearch";
+import { VectorStore } from "./base.js";
+import { Embeddings } from "../embeddings/base.js";
+import { Document } from "../document.js";
+
+export type ElasticSearchLibArgs = (
+  | { client: Client }
+  | { clientOptions: ClientOptions }
+) & {
+  indexName?: string;
+};
+
+interface VectorDocument {
+  vector: number[];
+  text: string;
+  metadata?: object;
+}
+
+/**
+ * Wrapper around Elasticsearch as a vector database.
+ */
+export class ElasticSearchStore extends VectorStore {
```

Comment:
In Python this seems to be called `ElasticVectorSearch`: https://python.langchain.com/docs/modules/data_connection/vectorstores/integrations/elasticsearch

Not sure if you prefer the synchronization or if this is a better name now?